src/builder.rs

```diff
@@ -52,8 +52,8 @@ impl Builder {
     }
 
     /// Which kind of network should be simulated.
-    pub fn ip_version(&mut self, value: IpVersion) -> &mut Self {
-        self.ip_version = value;
+    pub fn ip_network(&mut self, value: impl Into<IpNetwork>) -> &mut Self {
```

Comment:
This is a breaking change, requiring a version bump, so to echo what @marcbowes said... perhaps we can get closer to the desired state (at least at the public API) with this change to avoid subsequent breaking changes.